dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -115,6 +115,10 @@ def login
     provider = auth_hash.provider.to_s
     session[:sign_up_type] = provider
 
+    # For some providers, signups can happen without ever having hit the sign_up page, where
+    # our tracking data is usually populated, so do it here
+    SignUpTracking.begin_sign_up_tracking(session)
```

Comment:
Pulled this up from within `register_new_user` because we sometimes log sign-up results from within `User.from_omniauth`:

https://github.com/code-dot-org/code-dot-org/blob/b1ef6b05e5c3f4ff60f0318014b6500b8dda457a/dashboard/app/models/user.rb#L718